### ISSUE TITLE
Windows 11 ASIO devices not recognized

### ISSUE THREAD

**Issue by beisenreich** [LABSN/expyfun#471]
There appears to be an issue with using ASIO devices on windows 11 with the expyfun sound_controller _rtmixer. 

1) It should be noted that the current release of sounddevice has ASIO disabled as default (see [issue 496](https://github.com/spatialaudio/python-sounddevice/issues/496) ).  

2) Rolling back sounddevice to version 0.4.7 produces the following behavior:

i. running python(3.11) and importing expyfun in the initial call to _rtmixer.py, ASIO devices are not found by sounddevice.query_devices() (line 55)

ii. when running python(3.11) and importing sounddevice and rtmixer directly, sounddevice.query_devices() does return the ASIO devices, and playback through the ASIO soundcard works

Has anyone else encountered this issue?

**Comment by beisenreich** [comment]
It gets weirder. I did some more testing today and expyfun appears to be calling the right sounddevice library. I'm running version 0.4.7 which has the ASIO driver support included and calling sounddevice.__version__ from line 56 of _rtmixer.py returns version 0.4.7. I thought maybe the issue is with the import of rtmixer loading a different sounddevice library, this also returned back as version 0.4.7.  When I pull up the available hostapis expyfun can see from sounddevices I get a return of the ASIO but with devices=[]. 

Now the real weird part, if you import sounddevice before importing expyfun and initializing the ec. Everything works, version returns as 0.4.7, hostapis show ASIO driver support and devices return the Madiface USB. Thoughts?

**Comment by larsoner** [comment]
I'm not sure why the import order should matter. If you import `python-rtmixer` before `sounddevice` does it change the behavior?

**Comment by beisenreich** [comment]
Importing rtmixer before sounddevice does not impact the behavior. Instead it appears importing either rtmixer or sounddevice before importing or calling expyfun fixes the issue.

**Comment by larsoner** [comment]
:exploding_head: interesting... I wonder if we have code that sets a problematic env var or similar. Or maybe it's importing `pyglet` that is somehow problematic.

You could in theory try commenting out various parts of what `import expyfun` does (by digging into `expyfun/__init__.py` etc. and commenting stuff out) but for now I'm just relieved that there is some workaround for you!

**Comment by larsoner** [comment]
> In a Qt6 window, this code below will never run until you remove os.environ["SD_ENABLE_ASIO"] = "1" or place import sounddevice as sd before PyQt import. Took a week to realise import order affects sounddevice and ASIO.

Yikes! No idea why this would be the case. I know PyQt6 (and in most recent PySide6 as well) does some magic with `PyOS_Inputhook` but not sure that would be related

**Comment by hockinsk** [comment]
Yeah, it's weird, I wasted so much time trying to figure out why nothing ASIO could ever be found with sounddevice depsite several rebuilds, custom libportaudio.dll builds etc etc  and all it was was sounddevice needs to be imported first.
